module/zfs/zfs_replay.c

```diff
@@ -361,6 +361,10 @@ zfs_replay_create_acl(zfs_sb_t *zsb, lr_acl_create_t *lracl, boolean_t byteswap)
 			    lr->lr_uid, lr->lr_gid);
 		}
 
+		if (name == NULL) {
+			error = SET_ERROR(EINVAL);
+			goto bail;
+		}
```

Comment:
@luozhengzheng sorry to ask you to refresh this again.  But upon further reflection I think we should opt for a more comprehensive fix and move these checks in to `zfs_create()` and `zfs_mkdir()`.  While they aren't strictly a public interfaces the symbols are exported and therefore they really should be a better job of validating their arguments.  Something like this:

```diff
diff --git a/module/zfs/zfs_vnops.c b/module/zfs/zfs_vnops.c
index 6b5cda9..b9d9b58 100644
--- a/module/zfs/zfs_vnops.c
+++ b/module/zfs/zfs_vnops.c
@@ -1918,6 +1918,9 @@ zfs_mkdir(struct inode *dip, char *dirname, vattr_t *vap, 
            (vsecp || IS_EPHEMERAL(uid) || IS_EPHEMERAL(gid)))
                return (SET_ERROR(EINVAL));
 
+       if (dirname == NULL)
+               return (SET_ERROR(EINVAL));
+
        ZFS_ENTER(zsb);
        ZFS_VERIFY_ZP(dzp);
        zilog = zsb->z_log;
```